flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImplTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.channel;
+
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.flink.util.function.BiConsumerWithException;
+
+import org.junit.Test;
+
+import javax.annotation.Nonnull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import static org.apache.flink.runtime.checkpoint.channel.ChannelStateWriteRequestDispatcher.NO_OP;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * {@link ChannelStateWriteRequestExecutorImpl} test.
+ */
+public class ChannelStateWriteRequestExecutorImplTest {
+
+	@Test(expected = IllegalStateException.class)
+	public void testCloseAfterSubmit() throws Exception {
+		testCloseAfterSubmit(ChannelStateWriteRequestExecutor::submit);
+	}
+
+	@Test(expected = IllegalStateException.class)
+	public void testCloseAfterSubmitPriority() throws Exception {
+		testCloseAfterSubmit(ChannelStateWriteRequestExecutor::submitPriority);
+	}
+
+	@Test
+	public void testSubmitFailure() throws Exception {
+		testSubmitFailure(ChannelStateWriteRequestExecutor::submit);
+	}
+
+	@Test
+	public void testSubmitPriorityFailure() throws Exception {
+		testSubmitFailure(ChannelStateWriteRequestExecutor::submitPriority);
+	}
+
+	private void testCloseAfterSubmit(BiConsumerWithException<ChannelStateWriteRequestExecutor, ChannelStateWriteRequest, Exception> requestFun) throws Exception {
+		WorkerClosingDeque closingDeque = new WorkerClosingDeque();
+		ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(NO_OP, closingDeque);
+		closingDeque.setWorker(worker);
```

Comment:
The worker should start after created, otherwise even though we do not call close via `WorkerClosingDeque`, it can still encounter exception as did in `testSubmitFailure`.

Reply:
I didn't get your point here, can you please explain what do you mean?

Reply:
I mean it is better to add `worker.start()`, then it is easy to distinguish the conditions with below `testSubmitFailure`. `testSubmitFailure` is to test the impact without starting, and `testCloseAfterSubmit` is to test the impact withe explicit close after starting.